packages/editor/src/state/callbacks.ts

```diff
@@ -200,61 +164,99 @@ export const useResampleDiagram = () =>
+      return;
+    }
 
-    try {
-      const { onStart, onFinish } = await optimizer.resample(id, variation);
-      toast.dismiss(resamplingLoading);
-      onFinish
-        .then((info) => {
-          set(diagramWorkerState, (state) => ({
-            ...state,
-            optimizing: false,
-          }));
-          onUpdate(info);
-        })
-        .catch(onError);
-
-      await onStart;
-      set(diagramWorkerState, (state) => ({
-        ...state,
-        optimizing: true,
-      }));
+    toast.dismiss(resamplingLoading);
+    set(diagramWorkerState, (worker) => ({
+      ...worker,
+      optimizing: true,
+      stepSequenceId: resampleResult.value,
+    }));
+    set(diagramState, (diagram) => ({
+      ...diagram,
+      error:
+        diagram.error !== null && diagram.error.errorType !== "RuntimeError"
+          ? diagram.error
+          : null,
+    }));
 
-      const info = await optimizer.pollForUpdate();
-      if (info !== null) onUpdate(info);
-    } catch (error: unknown) {
-      onError(error);
-    }
+    // const diagram: Diagram = snapshot.getLoadable(diagramState)
+    //   .contents as Diagram;
+    // const id: string = snapshot.getLoadable(diagramWorkerState)
+    //   .contents as string;
+    // if (diagram.state === null) {
+    //   toast.error("Cannot resample uncompiled diagram");
+    //   return;
+    // }
+    // const variation = generateVariation();
+    // const resamplingLoading = toast.loading("Resampling...");
+    //
+    // const onError = (error: any) => {
+    //   _onError(error, set);
+    //   toast.dismiss(resamplingLoading);
+    // };
+    //
+    // const onUpdate = (info: UpdateInfo) => {
+    //   set(diagramState, (state) => ({
+    //     ...state,
+    //     metadata: { ...state.metadata, variation },
+    //     state: info.state,
+    //     // keep compile errors on resample, but clear runtime errors
+    //     error: state.error?.errorType === "RuntimeError" ? null : state.error,
+    //   }));
+    //   // update grid state too
+    //   set(diagramGridState, ({ gridSize }) => ({
+    //     variations: range(gridSize).map((i) =>
+    //       i === 0 ? variation : generateVariation(),
+    //     ),
+    //     gridSize,
+    //   }));
+    // };
+    //
+    // try {
+    //   const { onStart, onFinish } = await optimizer.resample(id, variation);
+    //   toast.dismiss(resamplingLoading);
+    //   onFinish
+    //     .then((info) => {
+    //       set(diagramWorkerState, (state) => ({
+    //         ...state,
+    //         optimizing: false,
+    //       }));
+    //       onUpdate(info);
+    //     })
+    //     .catch(onError);
+    //
+    //   await onStart;
+    //   set(diagramWorkerState, (state) => ({
+    //     ...state,
+    //     optimizing: true,
+    //   }));
+    //
+    //   const info = await optimizer.pollForUpdate();
+    //   if (info !== null) onUpdate(info);
+    // } catch (error: unknown) {
+    //   onError(error);
+    // }
```

Comment:
Is there a reason to keep this code here commented-out instead of just deleting it?

Reply:
Nope. I'll delete it : )